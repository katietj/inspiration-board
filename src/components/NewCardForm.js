import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      text: '',
      emoji: '',
      errorMessage: ''
    };
  }

  resetState = () => {
    this.setState({
      text: '',
      emoji: '',
      errorMessage: '',
    });
  }

  onFormChange = (event) => {
    const field = event.target.name;
    const value = event.target.value;

    const updatedState = {};
    updatedState[field] = value;
    this.setState(updatedState);
  }

  onSubmit = (event) => {
    event.preventDefault();
    this.props.onAddCardCallback({
      text: this.state.text,
      emoji: this.state.emoji,
    })

    console.log(event);
    this.resetState();
  }

  makeEmoji = () => {

    const emojiList = EMOJI_LIST.map((emojiWord) => {
      return (
      <option key={emojiWord} value={emojiWord}> {emoji.getUnicode(emojiWord)} </option>
    );
  });
  return emojiList
  }

  render() {
    return (
      <form onSubmit={this.onSubmit} name="new-card-form__form " id="new-card-form" className="new-card-form">
        <div>
          <label className="new-card-form__form-label" htmlFor="text-transform">Text</label>
          <input name="text" placeholder="text" onChange={this.onFormChange} value={this.state.text} />
        </div>
        <select name="emoji" onChange={this.onFormChange} value={this.state.emoji}>
        {this.makeEmoji()}
        </select>
        <input className="new-card-form__form-button" type="submit" name="submit" value="Add a Card" />
      </form>
    );
  }


}

NewCardForm.propTypes = {
  onAddCardCallback: PropTypes.func.isRequired,
};

export default NewCardForm;
