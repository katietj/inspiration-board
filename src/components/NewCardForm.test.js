import React from 'react';
import NewCardForm from './NewCardForm';
import { shallow } from 'enzyme';

describe('NewCardForm', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = shallow( <NewCardForm onAddCardCallback={() => {} } />);

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
