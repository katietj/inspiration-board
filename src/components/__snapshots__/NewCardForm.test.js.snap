// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches an existing snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewCardForm
    onAddCardCallback={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="new-card-form__form"
>
        <textarea
                className="new-card-form__form-textarea"
                name="text"
                onChange={[Function]}
                placeholder="Inspiration goes here!"
                value=""
        />
        <label
                className="new-card-form__form-label"
        >
                Pick an Emoji
        </label>
        <select
                className="new-card-form__form-select"
                name="emoji"
                onChange={[Function]}
                value=""
        >
                <option
                        value=""
                >
                         
                         
                </option>
                <option
                        value="heart_eyes"
                >
                         
                        😍
                         
                </option>
                <option
                        value="beer"
                >
                         
                        🍺
                         
                </option>
                <option
                        value="clap"
                >
                         
                        👏
                         
                </option>
                <option
                        value="sparkling_heart"
                >
                         
                        💖
                         
                </option>
                <option
                        value="heart_eyes_cat"
                >
                         
                        😻
                         
                </option>
                <option
                        value="dog"
                >
                         
                        🐶
                         
                </option>
        </select>
        <input
                className="new-card-form__form-button"
                name="submit"
                type="submit"
                value="Add a Card"
        />
</div>,
      "className": "new-card-form",
      "id": "new-card-form",
      "name": "new-card-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <textarea
            className="new-card-form__form-textarea"
            name="text"
            onChange={[Function]}
            placeholder="Inspiration goes here!"
            value=""
/>,
          <label
            className="new-card-form__form-label"
>
            Pick an Emoji
</label>,
          <select
            className="new-card-form__form-select"
            name="emoji"
            onChange={[Function]}
            value=""
>
            <option
                        value=""
            >
                         
                         
            </option>
            <option
                        value="heart_eyes"
            >
                         
                        😍
                         
            </option>
            <option
                        value="beer"
            >
                         
                        🍺
                         
            </option>
            <option
                        value="clap"
            >
                         
                        👏
                         
            </option>
            <option
                        value="sparkling_heart"
            >
                         
                        💖
                         
            </option>
            <option
                        value="heart_eyes_cat"
            >
                         
                        😻
                         
            </option>
            <option
                        value="dog"
            >
                         
                        🐶
                         
            </option>
</select>,
          <input
            className="new-card-form__form-button"
            name="submit"
            type="submit"
            value="Add a Card"
/>,
        ],
        "className": "new-card-form__form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-card-form__form-textarea",
            "name": "text",
            "onChange": [Function],
            "placeholder": "Inspiration goes here!",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pick an Emoji",
            "className": "new-card-form__form-label",
          },
          "ref": null,
          "rendered": "Pick an Emoji",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value=""
>
                 
                 
</option>,
              <option
                value="heart_eyes"
>
                 
                😍
                 
</option>,
              <option
                value="beer"
>
                 
                🍺
                 
</option>,
              <option
                value="clap"
>
                 
                👏
                 
</option>,
              <option
                value="sparkling_heart"
>
                 
                💖
                 
</option>,
              <option
                value="heart_eyes_cat"
>
                 
                😻
                 
</option>,
              <option
                value="dog"
>
                 
                🐶
                 
</option>,
            ],
            "className": "new-card-form__form-select",
            "name": "emoji",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  undefined,
                  " ",
                ],
                "value": "",
              },
              "ref": null,
              "rendered": Array [
                " ",
                undefined,
                " ",
              ],
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "heart_eyes",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "😍",
                  " ",
                ],
                "value": "heart_eyes",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "😍",
                " ",
              ],
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "beer",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "🍺",
                  " ",
                ],
                "value": "beer",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "🍺",
                " ",
              ],
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "clap",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "👏",
                  " ",
                ],
                "value": "clap",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "👏",
                " ",
              ],
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "sparkling_heart",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "💖",
                  " ",
                ],
                "value": "sparkling_heart",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "💖",
                " ",
              ],
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "heart_eyes_cat",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "😻",
                  " ",
                ],
                "value": "heart_eyes_cat",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "😻",
                " ",
              ],
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "dog",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "🐶",
                  " ",
                ],
                "value": "dog",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "🐶",
                " ",
              ],
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-card-form__form-button",
            "name": "submit",
            "type": "submit",
            "value": "Add a Card",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="new-card-form__form"
>
          <textarea
                    className="new-card-form__form-textarea"
                    name="text"
                    onChange={[Function]}
                    placeholder="Inspiration goes here!"
                    value=""
          />
          <label
                    className="new-card-form__form-label"
          >
                    Pick an Emoji
          </label>
          <select
                    className="new-card-form__form-select"
                    name="emoji"
                    onChange={[Function]}
                    value=""
          >
                    <option
                              value=""
                    >
                               
                               
                    </option>
                    <option
                              value="heart_eyes"
                    >
                               
                              😍
                               
                    </option>
                    <option
                              value="beer"
                    >
                               
                              🍺
                               
                    </option>
                    <option
                              value="clap"
                    >
                               
                              👏
                               
                    </option>
                    <option
                              value="sparkling_heart"
                    >
                               
                              💖
                               
                    </option>
                    <option
                              value="heart_eyes_cat"
                    >
                               
                              😻
                               
                    </option>
                    <option
                              value="dog"
                    >
                               
                              🐶
                               
                    </option>
          </select>
          <input
                    className="new-card-form__form-button"
                    name="submit"
                    type="submit"
                    value="Add a Card"
          />
</div>,
        "className": "new-card-form",
        "id": "new-card-form",
        "name": "new-card-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <textarea
              className="new-card-form__form-textarea"
              name="text"
              onChange={[Function]}
              placeholder="Inspiration goes here!"
              value=""
/>,
            <label
              className="new-card-form__form-label"
>
              Pick an Emoji
</label>,
            <select
              className="new-card-form__form-select"
              name="emoji"
              onChange={[Function]}
              value=""
>
              <option
                            value=""
              >
                             
                             
              </option>
              <option
                            value="heart_eyes"
              >
                             
                            😍
                             
              </option>
              <option
                            value="beer"
              >
                             
                            🍺
                             
              </option>
              <option
                            value="clap"
              >
                             
                            👏
                             
              </option>
              <option
                            value="sparkling_heart"
              >
                             
                            💖
                             
              </option>
              <option
                            value="heart_eyes_cat"
              >
                             
                            😻
                             
              </option>
              <option
                            value="dog"
              >
                             
                            🐶
                             
              </option>
</select>,
            <input
              className="new-card-form__form-button"
              name="submit"
              type="submit"
              value="Add a Card"
/>,
          ],
          "className": "new-card-form__form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-textarea",
              "name": "text",
              "onChange": [Function],
              "placeholder": "Inspiration goes here!",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Pick an Emoji",
              "className": "new-card-form__form-label",
            },
            "ref": null,
            "rendered": "Pick an Emoji",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value=""
>
                   
                   
</option>,
                <option
                  value="heart_eyes"
>
                   
                  😍
                   
</option>,
                <option
                  value="beer"
>
                   
                  🍺
                   
</option>,
                <option
                  value="clap"
>
                   
                  👏
                   
</option>,
                <option
                  value="sparkling_heart"
>
                   
                  💖
                   
</option>,
                <option
                  value="heart_eyes_cat"
>
                   
                  😻
                   
</option>,
                <option
                  value="dog"
>
                   
                  🐶
                   
</option>,
              ],
              "className": "new-card-form__form-select",
              "name": "emoji",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    undefined,
                    " ",
                  ],
                  "value": "",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  undefined,
                  " ",
                ],
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "heart_eyes",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "😍",
                    " ",
                  ],
                  "value": "heart_eyes",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "😍",
                  " ",
                ],
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "beer",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "🍺",
                    " ",
                  ],
                  "value": "beer",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "🍺",
                  " ",
                ],
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "clap",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "👏",
                    " ",
                  ],
                  "value": "clap",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "👏",
                  " ",
                ],
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "sparkling_heart",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "💖",
                    " ",
                  ],
                  "value": "sparkling_heart",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "💖",
                  " ",
                ],
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "heart_eyes_cat",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "😻",
                    " ",
                  ],
                  "value": "heart_eyes_cat",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "😻",
                  " ",
                ],
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "dog",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "🐶",
                    " ",
                  ],
                  "value": "dog",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "🐶",
                  " ",
                ],
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card-form__form-button",
              "name": "submit",
              "type": "submit",
              "value": "Add a Card",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
